modules/library/src/main/scala-3/zio/elasticsearch/ValidatedNewtypes.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.prelude.Assertion.isEmptyString
+import zio.prelude.{Newtype, NewtypeCustom}
+
+abstract class ValidatedNewtypes {
+  object IndexName extends NewtypeCustom[String] {
```

Comment:
Does it make sense to have separate objects for Index name and Routing. I would probably split this and remove `ValidatedNewtypes`.